src/gui/wizard/owncloudwizard.cpp

```diff
@@ -68,6 +70,41 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
     setPage(WizardCommon::Page_Result, _resultPage);
     setPage(WizardCommon::Page_WebView, _webViewPage);
 
+    // Add keyboard shortcut to allow the user to cancel wizard
+    auto *closeWindowAction = new QAction(this);
+    closeWindowAction->setShortcut(QKeySequence("Ctrl+W"));
+    connect(closeWindowAction, &QAction::triggered, this, &OwncloudWizard::reject);
+    addAction(closeWindowAction);
+
+    // Add keyboard shortcut to allow the user to quit
+    auto *quitAction = new QAction(this);
+    quitAction->setShortcut(QKeySequence("Ctrl+Q"));
+    connect(quitAction, &QAction::triggered, this, &QApplication::quit);
+    addAction(quitAction);
+
+    // Enable "Cancel" button
+    // (Will be re-disabled on second page of wizard)
+    // reject() is set to quit() if no accounts are set up
+    OwncloudWizard::setOption(QWizard::CancelButtonOnLeft);
+
+    if (AccountManager::instance()->accounts().isEmpty()) {
+        if (Utility::isMac()) {
+            setButtonText(WizardButton::CancelButton, tr("Quit"));
+        } else {
+            setButtonText(WizardButton::CancelButton, tr("Exit"));
+        }
+    }
+
+    // Make the application quit when the window is closed
+    // if no accounts are set up
+    // This setting will be reversed when the wizard is complete
+    if (AccountManager::instance()->accounts().isEmpty()) {
+        QApplication::setQuitOnLastWindowClosed(true);
```

Comment:
Do we really need to mess with that flag? I mean with the code you got in reject this looks redundant.

Reply:
You're right that this could probably be removed.